sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBufferCatalog.scala

```diff
@@ -598,6 +612,19 @@ class RapidsBufferCatalog(
       stream: Cuda.Stream): Unit = {
     val spillStoreMaxSize = spillStore.getMaxSize
     if (spillStoreMaxSize.isDefined) {
+      // Now only host memory store has a max limitation, and will spill its buffers to disk,
+      // Only spill to disk if allowed.
+      println(s"->allow task ${TaskContext.get.taskAttemptId()} spilling " +
+        s"to disk? $allowSpillingToDisk")
+      if (!allowSpillingToDisk) {
+        if (spillStore.currentSize + buffer.getMemoryUsedBytes > spillStoreMaxSize.get) {
+          throw new ExceedHostMaxSizeException(s"Go beyond the HostMemoryStore " +
+            s"max size: ${spillStoreMaxSize.get}")
+        } else {
+          return
```

Comment:
`return` looks benign here, but it is problematic in Scala. I would suggest not having this else at all, and wrapping the code starting line 630 with an else just to avoid it.